README.md

```diff
@@ -20,3 +20,13 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+# Solution
+
+Best case scenario is an already sorted array. This means that the first array in the permutations is the sorted array. However, generating the permutations takes O(n!) because there are n! permutations for an array of length n. This means that no matter the input, there is always a base complexitiy of O(n!). I now realize I could easily improve this by just checking if the permutation is sorted inside of my genPermutations function, so if you'd like me to do that let me know. We also need to linearly search through each permutation, taking O(n). For each permutation, we need to check if it is sorted, this is done by linearly checking each element one by one which means, parsing through the permutations takes $O(n\text{(from permutation array)}\cdot n \text{(from each element in permutation)})\rightarrow O(n^2)$, which grows slower than O(n!). Therefore, regardless of the input, my current program has a time complexity of $O(n!)$.
```

Comment:
Sorry, not following you here. There are $n!$ permutations, so how does going through the array of permutations take $n$ time? And if I understand correctly, you have to do this for each permutation, so you would have a non-constant factor for $n!$.